polygeist/tools/cgeist/driver.cc

```diff
@@ -918,75 +932,101 @@ parseOptimizationLevel(llvm::StringRef Arg) {
 
   for (int I = 0; I < argc; I++) {
     StringRef Ref(argv[I]);
-    if (Ref == "-Wl,--start-group")
+
+    if (Ref == "-Wl,--start-group") {
+      LinkOpts.push_back(argv[I]);
       LinkOnly = true;
-    if (!LinkOnly) {
-      if (Ref == "-fPIC" || Ref == "-c" || Ref.startswith("-fsanitize"))
-        LinkageArgs.push_back(argv[I]);
-      else if (Ref == "-L" || Ref == "-l") {
-        LinkageArgs.push_back(argv[I]);
-        I++;
-        LinkageArgs.push_back(argv[I]);
-      } else if (Ref.startswith("-L") || Ref.startswith("-l") ||
-                 Ref.startswith("-Wl"))
-        LinkageArgs.push_back(argv[I]);
-      else if (Ref == "-D" || Ref == "-I") {
-        MLIRArgs.push_back(argv[I]);
-        I++;
-        MLIRArgs.push_back(argv[I]);
-      } else if (Ref.startswith("-D")) {
-        MLIRArgs.push_back("-D");
-        MLIRArgs.push_back(&argv[I][2]);
-      } else if (Ref.startswith("-I")) {
-        MLIRArgs.push_back("-I");
-        MLIRArgs.push_back(&argv[I][2]);
-      } else if (Ref == "-fsycl-is-device") {
-        SYCLIsDevice = true;
-        MLIRArgs.push_back(argv[I]);
-      } else if (Ref == "--args") {
-        ClangOption = true;
-        MLIRArgs.push_back(argv[I]);
-      } else if (Ref.consume_front("-O") || Ref.consume_front("--optimize")) {
-        // If several flags are passed, we keep the last one.
-        OptimizationLevel = ExitOnErr(parseOptimizationLevel(Ref));
-        if (ClangOption)
-          MLIRArgs.push_back(argv[I]);
-        LinkageArgs.push_back(argv[I]);
-      } else if (Ref == "-g")
-        LinkageArgs.push_back(argv[I]);
-      else
-        MLIRArgs.push_back(argv[I]);
-    } else
-      LinkageArgs.push_back(argv[I]);
-
-    if (Ref == "-Wl,--end-group")
+      continue;
+    }
+
+    if (Ref == "-Wl,--end-group") {
+      LinkOpts.push_back(argv[I]);
       LinkOnly = false;
-  }
+      continue;
+    }
 
-  return {SYCLIsDevice, OptimizationLevel};
+    if (LinkOnly) {
+      LinkOpts.push_back(argv[I]);
+      continue;
+    }
+
+    if (Ref == "-fPIC" || Ref == "-c" || Ref.startswith("-fsanitize"))
+      LinkOpts.push_back(argv[I]);
+    else if (Ref == "-L" || Ref == "-l") {
+      LinkOpts.push_back(argv[I]);
+      I++;
+      LinkOpts.push_back(argv[I]);
+    } else if (Ref.startswith("-L") || Ref.startswith("-l") ||
+               Ref.startswith("-Wl"))
+      LinkOpts.push_back(argv[I]);
+    else if (Ref == "-D" || Ref == "-I") {
+      MLIROpts.push_back(argv[I]);
+      I++;
+      MLIROpts.push_back(argv[I]);
+    } else if (Ref.startswith("-D")) {
+      MLIROpts.push_back("-D");
+      MLIROpts.push_back(&argv[I][2]);
+    } else if (Ref.startswith("-I")) {
+      MLIROpts.push_back("-I");
+      MLIROpts.push_back(&argv[I][2]);
+    } else if (Ref == "-fsycl-is-device") {
+      CgeistOpts.setSYCLIsDevice();
+      if (ClangOption)
+        MLIROpts.push_back(argv[I]);
+    } else if (Ref == "-relaxed-aliasing") {
```

Comment:
No because those driver options are passed to the clang FE as `-relaxed-aliasing` (when -fno-strict-aliasing is set)  or nothing at all then `-fno-relaxed-aliasing` is not set. I'd like `cgeist` to use the same option name as the clang FE.

Reply:
Okey, good!